src/metabase/driver/postgres.clj

```diff
@@ -209,6 +213,94 @@
+
+                           [:case-expr :c.relkind
+                            [:inline "r"] [:inline "TEMPORARY TABLE"]
+                            [:inline "p"] [:inline "TEMPORARY TABLE"]
+                            [:inline "i"] [:inline "TEMPORARY INDEX"]
+                            [:inline "s"] [:inline "TEMPORARY SEQUENCE"]
+                            [:inline "v"] [:inline "TEMPORARY VIEW"]
+                            :else nil]]
+
+                          false ;; non system tables
+                          [:case-expr :c.relkind
+                           [:inline "r"] [:inline "TABLE"]
+                           [:inline "p"] [:inline "PARTITIONED TABLE"]
+                           [:inline "i"] [:inline "INDEX"]
+                           [:inline "P"] [:inline "PARTITIONED INDEX"]
+                           [:inline "S"] [:inline "SEQUENCE"]
+                           [:inline "v"] [:inline "VIEW"]
+                           [:inline "c"] [:inline "TYPE"]
+                           [:inline "f"] [:inline "FOREIGN TABLE"]
+                           [:inline "m"] [:inline "MATERIALIZED VIEW"]
+                           :else nil]
+                          :else nil]
+                         :type]
+                        [:d.description :description]]
+            :from      [[:pg_catalog.pg_namespace :n]
+                        [:pg_catalog.pg_class :c]]
+            :left-join [[:pg_catalog.pg_description :d] [:and [:= :c.oid :d.objoid] [:= :d.objsubid [:inline 0]]]
+                        [:pg_catalog.pg_class :dc]      [:and [:= :d.classoid :dc.oid] [:= :dc.relname [:inline "pg_class"]]]
+                        [:pg_catalog.pg_namespace :dn]  [:and [:= :dn.oid :dc.relnamespace] [:= :dn.nspname [:inline "pg_catalog"]]]]
+            :where     [:= :c.relnamespace :n.oid]
+            :order-by  [:type :schema :name]}
+    (seq schemas)
+    (sql.helpers/where [:in :n.nspname schemas])
+
+    (seq tables)
+    (sql.helpers/where [:in :c.relname tables])
+
+    (seq types)
+    (sql.helpers/where (into [:or] (map #(get table-type-clauses %) types))))
+   {:dialect :ansi}))
+
+(defn- get-tables
+  ;; have it as its own method for ease of testing
+  [conn schemas tables types]
+  (sql-jdbc.execute/reducible-query :postgres conn (get-tables-sql schemas tables types)))
+
+(defmethod driver/describe-database :postgres
+  [driver database]
+  {:tables
+   (sql-jdbc.execute/do-with-connection-with-options
+    driver
+    database
+    nil
+    (fn [^Connection conn]
+      (if-let [syncable-schemas (seq (sql-jdbc.describe-database/syncable-schemas-for-db database))]
+        (let [have-select-privilege? (sql-jdbc.describe-database/have-select-privilege-fn :postgres conn)]
+          (into #{} (comp (filter have-select-privilege?) (map #(dissoc % :type)))
+                (get-tables conn syncable-schemas nil
+                            ["TABLE" "PARTITIONED TABLE" "VIEW" "FOREIGN TABLE" "MATERIALIZED VIEW"])))
```

Comment:
I feel like the types are unneccessary argument to this function. This doesn't seem like it makes this code any more future-proof to me; I can't imagine why we would call `get-tables` with another set of types. Wdyt about removing the types parameter?